### ISSUE TITLE
Generation doesn't work for overridden main array

### ISSUE THREAD

**Issue by tpluscode** [yeoman/bower-requirejs#103]
Hi

I've got [Polymer](https://www.polymer-project.org/docs/start/getting-the-code.html#using-bower) installed from bower. Because it doesn't have a main defined in its bower.config I added the necessary override:

```
"overrides": {
    "polymer": {
      "main": [
        "polymer.js",
        "polymer.html",
        "layout.html"
      ]
    }
}
```

It is an array because I'm using [main-bower-files](https://github.com/ck86/main-bower-files) to copy what is needed (also uses wildcard for other components). When I run bower-requirejs it fails silently in a `catch` block. The error says `TypeError: Object polymer.js,polymer.html,layout.html has no method 'trim'`.

Is a fix possible? Also it would be nice if the error wasn't swallowed :)

**Comment by arthurvr** [comment]
> TypeError: Object polymer.js,polymer.html,layout.html has no method 'trim'

That problem isn't caused because it's in the `overrides`, we just don't handle arrays nicely. `main` may bower-wise be an array, but that's rjs-wise not correct. We probably need some kind of better error message, though.

> polymer.js,polymer.html,layout.html

For cases like this we probably just need to filter out the non-javascript files.

**Comment by delannee** [comment]
I have the same issue and don't know how to solve it. No workaround for this ?

**Comment by smoldaner** [comment]
bower-requirejs works already with multiple main files in bower.json. Why not just remove the trim() (IMHO it's not necessary)  in case of an override? Replacing 

var main = dep.main.trim(); 

with 

var main = dep.main;

in lib/build-config.js works like a charm.

**Comment by arthurvr** [comment]
@smoldaner A PR would definitely help to push this forward :)

**Comment by bezoerb** [comment]
LGTM
@smoldaner could you provide a testcase for the changes?

**Comment by smoldaner** [comment]
Test added. If you run the test against the the original code you will get the exception above.

**Comment by bezoerb** [comment]
Fixed by f09551f53120e2a584fda4533a8b9eb32b6b8fec

**Comment by patriziomunzi** [comment]
Hello, when will this fix be included in a official release.
This problem breaks the yeoman angularjs-require generator.
Thanks
